lxd/request/request.go

```diff
@@ -22,3 +24,23 @@ func CreateRequestor(ctx context.Context) *api.EventLifecycleRequestor {
 func SaveConnectionInContext(ctx context.Context, connection net.Conn) context.Context {
 	return context.WithValue(ctx, CtxConn, connection)
 }
+
+// GetCallerIdentityFromContext extracts the identity of the caller from the request context.
+// The identity is expected to be present and have a valid identifier. Otherwise, an error is returned.
+func GetCallerIdentityFromContext(ctx context.Context) (*identity.CacheEntry, error) {
+	requestor, err := GetRequestor(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	identity := requestor.CallerIdentity()
+	if identity == nil {
+		return nil, errors.New("Request context identity is missing")
+	}
+
+	if identity.Identifier == "" {
```

Comment:
@markylaing should this check be added to `requestor.CallerIdentity()` such that it returns nil if `identity.Identifier` is empty?

Do you think we need this `GetCallerIdentityFromContext` at all, and if so, should we be updating other call sites of `requestor.CallerIdentity()` to use it?